libass/ass.c

```diff
@@ -1455,18 +1455,22 @@ char *ass_load_file(ASS_Library *library, const char *fname, FileNameSource hint
         return NULL;
     }
 
-    const size_t CHUNK_SIZE = 16 * 1024;
-    size_t bufmax = 128 * CHUNK_SIZE;
-    char *buf = malloc(bufmax + 1); // one extra for terminating zero
+    char *buf = NULL;
+    size_t bufmax = 2 * 1024 * 1024;
+    ass_get_file_size(fp, &bufmax);
+    if (bufmax > SIZE_MAX - 1 || bufmax == 0)
+        goto fail;
+    buf = malloc(bufmax + 1); // one extra for terminating zero
     if (!buf)
         goto fail;
 
     size_t bytes_read = 0;
     while (true) {
-        size_t read = fread(buf + bytes_read, 1, CHUNK_SIZE, fp);
+        size_t maxread = bufmax - bytes_read;
```

Comment:
I suggest to ensure that `maxread` is always greater than `CHUNK_SIZE` to avoid suboptimal behavior at the file end.